symengine/rational.h

```diff
@@ -236,9 +239,12 @@ void get_num_den(const Rational &rat,
         const Ptr<RCP<const Integer>> &den);
 
 //! convenience creator from two longs
-inline RCP<const Number> rational(long n, long d) {
-    return Rational::from_two_ints(n, d);
-    }
+inline RCP<const Rational> rational(long n, long d)
+{
+    mpq_class rat(n, d);
+    rat.canonicalize();
+    return make_rcp<const Rational>(rat);
+}
```

Comment:
Ah I see --- the main difference is that `Rational::from_mpq` returns an Integer if `n/d` is an integer. While your new version will simply fail in Debug mode in that case (and undefined behavior in Release mode).

So we need to think how to best handle this.